tests/units/base.js

```diff
@@ -464,37 +475,69 @@ module( 'Base keyboard events', {
-
-    picker.$node.trigger({ type: 'keydown', keyCode: 46 })
-    ok( picker.get( 'open' ) === false, 'Closed after “alt. backspace”' )
-
-    picker.$node.trigger({ type: 'keydown', keyCode: 39 })
-    ok( picker.get( 'open' ) === true, 'Opened after arrow “right”' )
+    setTimeout(function () {
+        ok(picker.get('open') === true, 'Opened with key in')
+        picker.$node.blur()
+        $DOM.focusin()
+        setTimeout(function () {
+            ok(picker.get('open') === false, 'Closed with key out')
+            picker.$node.trigger({
+                type: 'keydown',
+                keyCode: 40
+            })
+            setTimeout(function () {
+                ok(picker.get('open') === true, 'Opened after arrow “down”')
+                picker.$node.trigger({
+                    type: 'keydown',
+                    keyCode: 27
+                })
+                setTimeout(function () {
+                    ok(picker.get('open') === false, 'Closed after “escape”')
+                    picker.$node.trigger({
+                        type: 'keydown',
+                        keyCode: 38
+                    })
+                    setTimeout(function () {
+                        ok(picker.get('open') === true, 'Opened after arrow “up”')
+                        picker.$node.trigger({
+                            type: 'keydown',
+                            keyCode: 8
+                        })
+                        setTimeout(function () {
+                            ok(picker.get('open') === false, 'Closed after “backspace”')
+                            picker.$node.trigger({
+                                type: 'keydown',
+                                keyCode: 37
+                            })
+                            setTimeout(function () {
+                                ok(picker.get('open') === true, 'Opened after arrow “left”')
+                                picker.$node.trigger({
+                                    type: 'keydown',
+                                    keyCode: 46
+                                })
+                                setTimeout(function () {
+                                    ok(picker.get('open') === false, 'Closed after “alt. backspace”')
+                                    picker.$node.trigger({
+                                        type: 'keydown',
+                                        keyCode: 39
+                                    })
+                                    setTimeout(function () {
+                                        ok(picker.get('open') === true, 'Opened after arrow “right”')
+                                        QUnit.start();
+                                    }, 500)
+                                }, 500)
+                            }, 500)
+                        }, 500)
+                    }, 500)
+                }, 500)
+            }, 500)
+        }, 500)
+    }, 500)
```

Comment:
Oh my 😯 lol

Reply:
Yeah, I suppose it's either this, or using lolex.

But on second thought, considering this is v3, I'd rather us spend less effort adding more dependencies.

Reply:
Unfortunately I know / knew no other direct solution as we uss QUnit 1 and did not want to touch more dependencies for this.